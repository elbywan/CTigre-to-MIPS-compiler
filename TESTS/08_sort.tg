var N := 100 in

type Tableau = array of int in

function TriInsertion (a:Tableau) =
  var j := 0
  and v := 0 in
  for i = 1 to N-1 do
    v := a[i]; j := i;
    while j > 0 & a[j-1] > v do  /* Attention : & doit etre paresseux */
      a[j] := a[j-1];
      j:= j-1
    done;
    a[j] := v
  done 
in

/* Un generateur basique de nombres vaguement dans le desordre */

function TableauAlea(init:int) : Tableau = 
  var a := Tableau [N] of 0 in
  function modulo(p,q) = p - (p/q)*q in
  var n := init in
  for i = 0 to N-1 do
    n := modulo(n*23,97);
    a[i] := n
  done;
  a
in

function AfficheTableau (a:Tableau) =
  for i = 0 to N-1 do
    print("a["); printint(i); print("]= "); printint(a[i]); print("\n")
  done
in

var a := TableauAlea(31) in
TriInsertion(a);
AfficheTableau(a)
