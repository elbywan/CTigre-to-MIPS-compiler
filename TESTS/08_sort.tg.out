a[0]= 1
a[1]= 2
a[2]= 3
a[3]= 4
a[4]= 5
a[5]= 6
a[6]= 6
a[7]= 7
a[8]= 8
a[9]= 9
a[10]= 10
a[11]= 11
a[12]= 12
a[13]= 13
a[14]= 14
a[15]= 15
a[16]= 16
a[17]= 17
a[18]= 18
a[19]= 19
a[20]= 20
a[21]= 21
a[22]= 22
a[23]= 23
a[24]= 24
a[25]= 25
a[26]= 26
a[27]= 27
a[28]= 28
a[29]= 29
a[30]= 30
a[31]= 31
a[32]= 32
a[33]= 33
a[34]= 34
a[35]= 34
a[36]= 35
a[37]= 36
a[38]= 37
a[39]= 38
a[40]= 39
a[41]= 40
a[42]= 41
a[43]= 41
a[44]= 42
a[45]= 43
a[46]= 44
a[47]= 45
a[48]= 46
a[49]= 47
a[50]= 48
a[51]= 49
a[52]= 50
a[53]= 51
a[54]= 52
a[55]= 53
a[56]= 54
a[57]= 55
a[58]= 56
a[59]= 57
a[60]= 58
a[61]= 59
a[62]= 60
a[63]= 61
a[64]= 62
a[65]= 63
a[66]= 64
a[67]= 65
a[68]= 66
a[69]= 67
a[70]= 68
a[71]= 69
a[72]= 70
a[73]= 70
a[74]= 71
a[75]= 72
a[76]= 73
a[77]= 74
a[78]= 75
a[79]= 76
a[80]= 77
a[81]= 78
a[82]= 79
a[83]= 80
a[84]= 81
a[85]= 82
a[86]= 83
a[87]= 84
a[88]= 85
a[89]= 86
a[90]= 87
a[91]= 88
a[92]= 89
a[93]= 90
a[94]= 91
a[95]= 92
a[96]= 93
a[97]= 94
a[98]= 95
a[99]= 96

